test/e2e/tests/reset-account.spec.js

```diff
@@ -0,0 +1,70 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Reset account', function () {
+  // ////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  // When user get stuck with pending transactions, one can reset the account by clicking the 'Reset account' //
+  // button in settings, advanced tab. This functionality will clear all the send transactions history.       //
+  // Note that the receive transactions history will be kept and it only only affects the current network.    //
+  // ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+  it('User can reset account via the advanced setting tab, ', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: convertToHexValue(25000000000000000000),
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withTransactionControllerMultipleTransactions()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        // Login and accept all transactions
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+        let transactionSize = 0;
+        while (transactionSize < 4) {
```

Comment:
Can we add a `confirmed` transactions to the TransactionController instead? This way we do not need to confirm the transactions in the test.

Reply:
Can we also add a confirmed transaction to the IncomingTransactionsController? This will allow us to check that the `Receive` transaction remains

Reply:
Using existing `withTransactionControllerApprovedTransaction() ` now, please check, thanks.
For the incoming transactions, as we are using localhost with ganache in e2s tests, incoming transactions don't show in the activity list, cannot check it on UI currently.

Reply:
Added a new method for incoming transaction, please review, thanks.